fast-vm

```diff
@@ -966,9 +982,38 @@ case "$1" in
 				pmsg $P_ERROR "$3 not executable\n" $vm_number
 			fi
 		else
-			ssh -o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "root@192.168.$SUBNET_NUMBER.$vm_number"
+			use_sshkey_if_exists
+			ssh $FASTVM_USE_SSHKEY -o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "root@192.168.$SUBNET_NUMBER.$vm_number"
 		fi
 		;;
+	scp|keydist)
+		vm_number="$2"
+		if [ -z "$vm_number" ]; then pmsg $P_ERROR "VM number missing\n"; usage ssh; fi
+		if [[ $vm_number = *[\ ]* ]]; then handle_multiple_requests $@;	fi
+
+		check_vm_number "$vm_number" "inactive"
+		update_access_time "$vm_number"
+
+		wait_for_ssh "$vm_number"
+		if [ "$1" == "keydist" ]; then
+			ssh-copy-id -i "$FASTVM_USER_CONF_DIR/id_fastvm" \
+				-o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no \
+				"root@192.168.$SUBNET_NUMBER.$vm_number" | sed "s/with:.*/with: $0 ssh $vm_number/"
+		else
+			# if no destination is given, that is ok
+			sourcepath="$3"
+			destpath="$4"
+			use_sshkey_if_exists
+			scp $FASTVM_USE_SSHKEY -o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$sourcepath" "root@192.168.$SUBNET_NUMBER.$vm_number:$destpath"
+		fi
+		;;
+	keygen)
```

Comment:
Is it really necessary to make this as another function?
To make this more convenient for user it would make more sense to me to generate the SSH key when user tries to do the `fast-vm ssh`, `fast-vm scp` or `fast-vm keydist` and there is no SSH key found.

Reply:
You're right, in this version I have made it automatic, thanks.